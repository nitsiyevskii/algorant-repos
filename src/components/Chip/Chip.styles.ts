import { createStyles } from 'src/utils/createStyles';

export const useStyles = createStyles((theme) => ({
  chip: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    borderRadius: 20,
    backgroundColor: theme.colors.background.secondary,
    borderWidth: 1,
    borderColor: theme.colors.separator.primary,
  },
  chipActive: {
    backgroundColor: theme.colors.core.primary,
    borderColor: theme.colors.core.primary,
  },
  chipText: {
    fontSize: theme.fonts.size.sm,
    color: theme.colors.text.primary,
    fontWeight: theme.fonts.weight.medium,
  },
  chipTextActive: {
    color: theme.colors.button.text,
  },
}));

