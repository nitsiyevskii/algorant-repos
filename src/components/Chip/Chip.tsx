import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import { useStyles } from './Chip.styles';

interface ChipProps {
  label: string;
  isSelected: boolean;
  onPress: () => void;
  testID?: string;
}

export const Chip: React.FC<ChipProps> = ({
  label,
  isSelected,
  onPress,
  testID,
}) => {
  const styles = useStyles();

  return (
    <TouchableOpacity
      style={[styles.chip, isSelected && styles.chipActive]}
      onPress={onPress}
      testID={testID}
    >
      <Text style={[styles.chipText, isSelected && styles.chipTextActive]}>
        {label}
      </Text>
    </TouchableOpacity>
  );
};

