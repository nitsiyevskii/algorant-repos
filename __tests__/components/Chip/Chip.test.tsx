import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Chip } from 'src/components/Chip/Chip';

describe('Chip', () => {
  const mockOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with label', () => {
    const { getByText } = render(
      <Chip label="TypeScript" isSelected={false} onPress={mockOnPress} />
    );
    expect(getByText('TypeScript')).toBeTruthy();
  });

  it('renders as selected', () => {
    const { toJSON } = render(
      <Chip label="JavaScript" isSelected={true} onPress={mockOnPress} />
    );
    expect(toJSON()).toBeTruthy();
  });

  it('renders as unselected', () => {
    const { toJSON } = render(
      <Chip label="Python" isSelected={false} onPress={mockOnPress} />
    );
    expect(toJSON()).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const { getByTestId } = render(
      <Chip
        label="Go"
        isSelected={false}
        onPress={mockOnPress}
        testID="chip-go"
      />
    );

    const chip = getByTestId('chip-go');
    fireEvent.press(chip);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot when unselected', () => {
    const { toJSON } = render(
      <Chip label="Rust" isSelected={false} onPress={mockOnPress} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot when selected', () => {
    const { toJSON } = render(
      <Chip label="Swift" isSelected={true} onPress={mockOnPress} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

