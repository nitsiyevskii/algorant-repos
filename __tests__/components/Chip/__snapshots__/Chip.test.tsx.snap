// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip matches snapshot when selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#007AFF",
      "borderColor": "#007AFF",
      "borderRadius": 20,
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "500",
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Swift
  </Text>
</View>
`;

exports[`Chip matches snapshot when unselected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#E5E5EA",
      "borderColor": "#C6C6C8",
      "borderRadius": 20,
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "500",
        },
        false,
      ]
    }
  >
    Rust
  </Text>
</View>
`;
